job.go

```diff
@@ -45,6 +45,7 @@ type jobFunction struct {
 	function          any                // task's function
 	parameters        []any              // task's function parameters
 	parametersLen     int                // length of the passed parameters
+	lockKey           string             // key of the distributed lock
 	name              string             // nolint the function name to run
```

Comment:
i'd like to change the naming a bit here so this isn't scoped to only distributed locking
```suggestion
	jobName           string             // key of the distributed lock
	funcName          string             // the name of the function - e.g. main.func1
```